core/src/main/java/org/apache/iceberg/avro/InternalReader.java

```diff
@@ -76,6 +76,15 @@ public void setSchema(Schema schema) {
     initReader();
   }
 
+  @Override
+  public void setCustomTypes(
```

Comment:
Because the `InternalReader` is no longer created by `ManifestReader`, the custom types now need to be passed to the read builder and forwarded to the reader here. Custom type support will need to be implemented for Parquet as well.